### DISCUSSION TITLE
Numbered markers

### DISCUSSION THREAD

**Discussion by jorge-plaza** [geoman-io/leaflet-geoman#1450] [Questions & Help] [answered]
Hi,

I would like to be able to create a serie of numbered markers like in the screenshot.
I'm using Vue and using a reference in the marker icon definition, but it draws only the first value
```ts
const list = ref([1])
const options = {
  markerStyle: {
    icon: L.divIcon({html: `${list.value.length}`})
  }
}
map.pm.enableDraw('Marker', options)
```
I've managed to do it with just with leaflet, but I would like to integrate with this tool.
Thanks for any help 😃 
![image](https://github.com/geoman-io/leaflet-geoman/assets/39694654/887675b9-0c0b-4e1a-ba43-9422dd501aae)

**Comment by Falke-Design** [comment, marked as answer]
After placing a Marker and continuing with drawing, the event `pm:drawstart` will be fired on the map. Then you can set the new icon over the options and apply the new style.

```
map.on('pm:drawstart', (e)=>{
   if(e.shape === 'Marker') {
      map.pm.setGlobalOptions({markerStyle : { icon: YOUR_DIV_ICON } });
      map.pm.Draw.Marker.setStyle();
   }
});
```
I haven't tested this code but it should help you enough

**Reply by mbajur** [reply]
This code applies YOUR_DIV_ICON options to all the `Marker` markers on the map - which in my case overwrites `draggable` setting for all of them and i have to write a really messy code to overcome that. Is there any way to update markerStyle option only on the marker that is currently being drawn instead of operating on all already drawn?
